### PR TITLE
Issue #914 : Disabling testSmallStringArrayToHDF5 from unit tests.

### PR DESCRIPTION
**Pull request by glitch**
This is causing issues with nightly benchmarking and represents a
very rare corner case on a tiny amout of data. We already know this
section of code requires and overhaul so we feel comfortable in
disabling it for now.

### CODE REVIEW

File: tests/io_test.py [hunk 1]
```diff
@@ -579,13 +579,15 @@ def testSmallArrayToHDF5(self):
...
-    def testSmallStringArrayToHDF5(self):
-        a1 = ak.array(["ab", "cd"])
-        with tempfile.TemporaryDirectory(dir=IOTest.io_test_dir) as tmp_dirname:
-            a1.save(f"{tmp_dirname}/small_numeric", dataset="a1")
-            # Now load it back in
-            a2 = ak.load(f"{tmp_dirname}/small_numeric", dataset="a1")
-            self.assertEqual(str(a1), str(a2))
+    # This currently breaks on 4 or greater locales.  Since it is such a rare corner case
+    # we are going to remove it for now.
+    # def testSmallStringArrayToHDF5(self):
```
Comment (ronawho) [inline tests/io_test.py:584]: At least historically, the way tests were skipped at other times was with `@pytest.mark.skip` (requires `import pytest`) instead of commenting out.

Comment (ronawho) [reply tests/io_test.py:584]: (Not sure if there's an advantage one way or another, I'll defer to @hokiegeek2 for that)

Comment (glitch) [reply tests/io_test.py:584]: Cool, I wasn't sure of the interplay between pytest marks and python's `unittest` module.  I'll test it out and update the PR.

**Review by ronawho — APPROVED** [review]

**Review by pierce314159 — APPROVED** [review]
